src/compiler/scala/tools/nsc/backend/jvm/BTypes.scala

```diff
@@ -1137,7 +1137,11 @@ object BTypes {
                               warning: Option[ClassInlineInfoWarning]) {
     lazy val methodInfosSorted: IndexedSeq[((String, String), MethodInlineInfo)] = {
       val result = new Array[((String, String), MethodInlineInfo)](methodInfos.size)
-      methodInfos.copyToArray(result)
+      var i = 0
+      methodInfos.foreachEntry { (ownerAndName, info) =>
+        result(i) = (ownerAndName, info)
```

Comment:
I realise now that I've just pushed the `Tuple` allocation here instead of `immutabe.HashMap##getPayload`. More rejigging for efficiency.

  - We could use a `LinkedHashMap` when [building methodinfos](https://github.com/scala/scala/blob/ec3ffb77f760e75474aad355387cbffb789645b3/src/compiler/scala/tools/nsc/backend/jvm/BTypesFromSymbols.scala#L568)
  - Or just a `Seq`, and lazily convert it to a `Map` in `CallGraph` if/when the optimizer actually is asked to do intra-procedural opts.

Reply:
I was :eyes: on the tuple from iterating in copyToArray.

Reply:
Sine we are interested on iterating through the instances in a specific order, we could change the `InlineInfo` map to be a to use a `TreeMap`? That could give us the iteration for free.

This would add cost, of course, for looking up values per entry; which would go from "arrayish hash" to logartithmic. Is that a major use-case for this `Map` object?

Reply:
A better alternative to the `TreeMap` could be to implement a `ArraySortedMap`, an implementation of `SortedMap` that is backed by two arrays (keys and values) in which the first one is always sorted. 
This would have the benefit that, since upon `InlineInfo` we have a fair idea of the maximum size (the length of `method`), the allocations here would be just those two arrays.